<a href="https://colab.research.google.com/github/DeVcB13d/Python-Lab/blob/main/Decision_Problems.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Decision Problems


1. Program to find the largest of two numbers

In [ ]:
a = int(input("Enter number A :"))
b = int(input("Enter number B :"))
if a > b :
  print(a,">",b)
elif a < b :
  print(b,">",a)
else:
  print(a,"=",b)

Enter number A :4
Enter number B :3
4 > 3



<style type="text/css">
.tg  {border-collapse:collapse;border-color:#9ABAD9;border-spacing:0;}
.tg td{background-color:#EBF5FF;border-color:#9ABAD9;border-style:solid;border-width:1px;color:#444;
  font-family:Arial, sans-serif;font-size:14px;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{background-color:#409cff;border-color:#9ABAD9;border-style:solid;border-width:1px;color:#fff;
  font-family:Arial, sans-serif;font-size:14px;font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-hmp3{background-color:#D2E4FC;text-align:left;vertical-align:top}
.tg .tg-0lax{text-align:left;vertical-align:top}
</style>
<table class="tg">
<thead>
  <tr>
    <th class="tg-0lax">No:</th>
    <th class="tg-0lax">Input Values</th>
    <th class="tg-0lax">Expected outcome</th>
    <th class="tg-0lax">Actual outcome</th>
    <th class="tg-0lax">Pass/Fail</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-hmp3">1</td>
    <td class="tg-hmp3">a = 1 , b =2</td>
    <td class="tg-hmp3">2</td>
    <td class="tg-hmp3">2</td>
    <td class="tg-hmp3">Pass</td>
  </tr>
  <tr>
    <td class="tg-0lax">2</td>
    <td class="tg-0lax">a = 4,b = 2</td>
    <td class="tg-0lax">4</td>
    <td class="tg-0lax">4</td>
    <td class="tg-0lax">Pass</td>
  </tr>
  <tr>
    <td class="tg-hmp3">3</td>
    <td class="tg-hmp3">a = 1 , b = 1</td>
    <td class="tg-hmp3">equal</td>
    <td class="tg-hmp3">equal</td>
    <td class="tg-hmp3">Pass</td>
  </tr>
</tbody>
</table>




2. Program to find the smallest of 3 numbers

In [ ]:
a = int(input("Enter number 1 :"))
b = int(input("Enter number 2 :"))
c = int(input("Enter number 3 :"))
if a < b and a < c :
  print(a," is the smallest")
elif b < a and b < c :
  print(b," is the smallest")
elif c < a and c < b :
  print(c," is the smallest")
else:
  print("All are equal")

Enter number 1 :0
Enter number 2 :0
Enter number 3 :40
All are equal


**Test Cases**
<table style="width: 539px; height: 194px; border-color: Black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;No:</td>
<td style="height: 23px; width: 211.337px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 92.7px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 90.4375px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;1</td>
<td style="height: 23px; width: 211.337px;">&nbsp;a = 2 , b=4, c = 5</td>
<td style="height: 23px; width: 92.7px;">&nbsp;2 is the smallest</td>
<td style="height: 23px; width: 90.4375px;">&nbsp;2 is the smallest</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46.325px;">&nbsp;2</td>
<td style="height: 23.5px; width: 211.337px;">&nbsp;a = 7 , b = 9 , c = 0</td>
<td style="height: 23.5px; width: 92.7px;">&nbsp;0 is the smallest</td>
<td style="height: 23.5px; width: 90.4375px;">&nbsp;0 is the smallest</td>
<td style="height: 23.5px; width: 97.2px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;3</td>
<td style="height: 23px; width: 211.337px;">&nbsp;a = 2 , b= 2 ,c =40</td>
<td style="height: 23px; width: 92.7px;">&nbsp;2 is the smallest</td>
<td style="height: 23px; width: 90.4375px;">&nbsp;All are equal</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;4</td>
<td style="height: 23px; width: 211.337px;">&nbsp;a = 1, b = 1 , c =1</td>
<td style="height: 23px; width: 92.7px;">&nbsp;All are equal</td>
<td style="height: 23px; width: 90.4375px;">&nbsp;All are equal</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Pass</td>
</tr>
</tbody>
</table>
<!-- DivTable.com -->

3. Program to find distance between two points on a number line



In [ ]:
x1 = int(input("Enter the 1st coordinate: "))
x2 = int(input("Enter the 2nd coordinate: "))
if x1 > x2 :
  print("Distance = ",x1-x2)
else:
  print("Distance = ",x2-x1)

Enter the 1st coordinate: -2
Enter the 2nd coordinate: 3
Distance =  5


**Test cases**
<table style="width: 539px; height: 194px; border-color: Black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;No:</td>
<td style="height: 23px; width: 211.337px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 92.7px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 90.4375px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;1</td>
<td style="height: 23px; width: 211.337px;">&nbsp;x1 = 3, x2 = 2</td>
<td style="height: 23px; width: 92.7px;">Distance = 1</td>
<td style="height: 23px; width: 90.4375px;">Distance = 1</td>
<td style="height: 23px; width: 97.2px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46.325px;">&nbsp;2</td>
<td style="height: 23.5px; width: 211.337px;">x1 = 2, x2 = 3</td>
<td style="height: 23.5px; width: 92.7px;">Distance = 1</td>
<td style="height: 23.5px; width: 90.4375px;">Distance = 1</td>
<td style="height: 23.5px; width: 97.2px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46.325px;">&nbsp;3</td>
<td style="height: 23px; width: 211.337px;">x1 = -2, x2 = 3</td>
<td style="height: 23px; width: 92.7px;">Distance = 5</td>
<td style="height: 23px; width: 90.4375px;">Distance = 5</td>
<td style="height: 23px; width: 97.2px;">Pass</td>
</tr>
</tbody>
</table>
<!-- DivTable.com -->

4. Program to check whether the input number is a 2 digit number
or not

In [ ]:
a = int(input("Enter a number: "))
if a >= 10 and a <= 99 :
  print(a," is 2 digit")
else:
  print(a," is not 2 digit")

Enter a number: 999
999  is not 2 digit


**Test cases**
<table style="width: 539px; height: 194px; border-color: Black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;No:</td>
<td style="height: 23px; width: 169px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 111.938px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 108.062px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 97px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;1</td>
<td style="height: 23px; width: 169px;">a = 1</td>
<td style="height: 23px; width: 111.938px;">1 is not 2 digit</td>
<td style="height: 23px; width: 108.062px;">1 is not 2 digit</td>
<td style="height: 23px; width: 97px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46px;">&nbsp;2</td>
<td style="height: 23.5px; width: 169px;">a = 99</td>
<td style="height: 23.5px; width: 111.938px;">99 is 2 digit</td>
<td style="height: 23.5px; width: 108.062px;">99 is 2 digit</td>
<td style="height: 23.5px; width: 97px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;3</td>
<td style="height: 23px; width: 169px;">a = 999</td>
<td style="height: 23px; width: 111.938px;">999 is not 2 digit</td>
<td style="height: 23px; width: 108.062px;">999 is not 2 digit</td>
<td style="height: 23px; width: 97px;">Pass</td>
</tr>
</tbody>
</table>
<!-- DivTable.com -->

5. Program to check whether a number is positive, negative or
zero

In [3]:
a = int(input("Enter a number:"))
if a < 0 :
  print(a," is negative")
elif a > 0:
  print(a," is positive")
else:
  print(a,"is zero")

Enter a number:0
0 is zero


6. Program to whether a number is a multiple of another

In [ ]:
n = int(input("Enter number  :"))
m = int(input("Enter number  :"))
if n%m == 0:
  print(n," is a multiple of ",m)
else:
  print("They are not multiples")

Enter number  :6
Enter number  :3
6  is a multiple of  3


**Test Cases**
<table style="width: 539px; height: 194px; border-color: Black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;No:</td>
<td style="height: 23px; width: 170.663px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 107.337px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 108px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 97px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;1</td>
<td style="height: 23px; width: 170.663px;">a = 5</td>
<td style="height: 23px; width: 107.337px;">5 is positive</td>
<td style="height: 23px; width: 108px;">5 is positive</td>
<td style="height: 23px; width: 97px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46px;">&nbsp;2</td>
<td style="height: 23.5px; width: 170.663px;">a = -10</td>
<td style="height: 23.5px; width: 107.337px;">-10 is negative</td>
<td style="height: 23.5px; width: 108px;">-10 is negative</td>
<td style="height: 23.5px; width: 97px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;3</td>
<td style="height: 23px; width: 170.663px;">a = 0</td>
<td style="height: 23px; width: 107.337px;">0 is zero</td>
<td style="height: 23px; width: 108px;">0 is zero</td>
<td style="height: 23px; width: 97px;">Pass</td>
</tr>
</tbody>
</table>

7 Program to find the smallest digit in a 2-digit number

In [7]:
d = int(input("Enter a 2 digit number"))
d1 = d%10
d2 = d//10
if d1 > d2 :
  print(d2,"is the smallest digit in ",d)
elif d1 < d2 :
  print(d1,"is the smallest digit in ",d)
else:
  print("Equal digits")

Enter a 2 digit number78
7 is the smallest digit in  78


**Test Cases**
<table style="width: 596px; height: 194px; border-color: black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;No:</td>
<td style="height: 23px; width: 112px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 147px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 186.763px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 89.2375px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;1</td>
<td style="height: 23px; width: 112px;">d = 54</td>
<td style="height: 23px; width: 147px;">4 is the smallest digit</td>
<td style="height: 23px; width: 186.763px;">4 is the smallest digit</td>
<td style="height: 23px; width: 89.2375px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46px;">&nbsp;2</td>
<td style="height: 23.5px; width: 112px;">d = 46</td>
<td style="height: 23.5px; width: 147px;">4 is the smallest digit</td>
<td style="height: 23.5px; width: 186.763px;">4 is the smallest digit</td>
<td style="height: 23.5px; width: 89.2375px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;3</td>
<td style="height: 23px; width: 112px;">d = 78</td>
<td style="height: 23px; width: 147px;">7 is the smallest digit</td>
<td style="height: 23px; width: 186.763px;">7 is the smallest digit</td>
<td style="height: 23px; width: 89.2375px;">Pass</td>
</tr>
</tbody>
</table>

8 Program to find the nature of the solution of a quadratic equation

In [14]:
a = int(input("Enter coefficient of x^2: "))
b = int(input("Enter coefficient of x: "))
c = int(input("Enter constant term: "))
det = (b**2) - (4*a*c)
if det > 0 : 
  print("2 different solutions ")
elif det < 0 :
  print("Only imaginary solutions ")
else:
  print("2 equal solutions")

Enter coefficient of x^2: 4
Enter coefficient of x: 0
Enter constant term: 0
2 equal solutions


**Test Cases**
<table style="width: 539px; height: 194px; border-color: Black;">
<tbody>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;No:</td>
<td style="height: 23px; width: 148px;">&nbsp;Input Values</td>
<td style="height: 23px; width: 148px;">
<p>Expected outcome&nbsp;</p>
</td>
<td style="height: 23px; width: 144px;">&nbsp;Actual Outcome</td>
<td style="height: 23px; width: 44px;">&nbsp;Pass/Fail</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;1</td>
<td style="height: 23px; width: 148px;">&nbsp;a = 1, b = 2, c = 1</td>
<td style="height: 23px; width: 148px;">2 equal solutions</td>
<td style="height: 23px; width: 144px;">2 equal solutions</td>
<td style="height: 23px; width: 44px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23.5px;">
<td style="height: 23.5px; width: 46px;">&nbsp;2</td>
<td style="height: 23.5px; width: 148px;">&nbsp;a = 1, b = -1, c = 1</td>
<td style="height: 23.5px; width: 148px;">imaginary solutions</td>
<td style="height: 23.5px; width: 144px;">imaginary solutions</td>
<td style="height: 23.5px; width: 44px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;3</td>
<td style="height: 23px; width: 148px;">&nbsp;a = 1, b= 5, c =1</td>
<td style="height: 23px; width: 148px;">2 different solutions</td>
<td style="height: 23px; width: 144px;">2 different solutions</td>
<td style="height: 23px; width: 44px;">&nbsp;Pass</td>
</tr>
<tr style="height: 23px;">
<td style="height: 23px; width: 46px;">&nbsp;4</td>
<td style="height: 23px; width: 148px;">&nbsp;a = 4, b = 0, c =0</td>
<td style="height: 23px; width: 148px;">2 equal solutions</td>
<td style="height: 23px; width: 144px;">2 equal solutions</td>
<td style="height: 23px; width: 44px;">&nbsp;Pass</td>
</tr>
</tbody>
</table>